rapid/src/main/java/com/vrg/rapid/MembershipService.java

```diff
@@ -494,9 +509,36 @@ int getMembershipSize() {
      * Shuts down all the executors.
      */
     void shutdown() {
+        leave();
```

Comment:
Given that leave() could throw an exception, it's better to leave the rest of the methods in the finally part of a try-finally block